src/drive/mobile/modules/mediaBackup/duck/index.js

```diff
@@ -34,9 +37,95 @@ const currentMediaUpload = (media, uploadCounter, totalUpload) => ({
   }
 })
 
-const getDirID = async path => {
-  const { _id } = await cozy.client.files.createDirectoryByPath(path, false)
-  return _id
+const getFoldersWithReference = async reference => {
+  const referencedFolders = await cozy.client.fetchJSON(
+    'POST',
+    '/files/_find',
+    {
+      selector: {
+        referenced_by: {
+          $elemMatch: {
+            type: reference
+          }
+        }
+      },
+      sort: [{ created_at: 'desc' }]
+    }
+  )
+
+  return referencedFolders.filter(
+    folder => /^\/\.cozy_trash/.test(folder.attributes.path) === false
+  )
+}
+
+const addFolderReference = (folderId, reference) =>
+  cozy.client.fetchJSON(
+    'POST',
+    `/files/${folderId}/relationships/referenced_by`,
+    {
+      data: [
+        {
+          type: reference,
+          id: reference
+        }
+      ]
+    }
+  )
+
+const ensureFolderWithReferenceExists = async (path, reference) => {
+  const referencedFolders = await getFoldersWithReference(reference)
+
+  if (referencedFolders.length) {
+    return referencedFolders[0]
+  } else {
+    const dir = await cozy.client.files.createDirectoryByPath(path, false)
+    await addFolderReference(dir._id, reference)
+    return dir
```

Comment:
try / catch ?

Reply:
What would we do with a caught exception here? I don't think we can recover from an error here, so I feel like the best place to catch the error is in the main backup function where there [already is a try/catch block](https://github.com/y-lohse/cozy-drive/blob/magic-folders/src/drive/mobile/modules/mediaBackup/duck/index.js#L160).